pkg/controller/highperf/highperf_controller_test.go

```diff
@@ -0,0 +1,88 @@
+package highperf
+
+import (
+	"context"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	fakeClient "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+	fakeNetwork "k8s.io/cloud-provider-gcp/crd/client/network/clientset/versioned/fake"
+	networkinformer "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/network/v1"
+	alphanetworkinformer "k8s.io/cloud-provider-gcp/crd/client/network/informers/externalversions/network/v1alpha1"
+	"k8s.io/cloud-provider-gcp/pkg/controller/testutil"
+	"k8s.io/cloud-provider-gcp/providers/gce"
+	"k8s.io/component-base/metrics/prometheus/controllers"
+	"testing"
+	"time"
+)
+
+type testHighPerfController struct {
+	stop          context.CancelFunc
+	ctx           context.Context
+	gceCloud      *gce.Cloud
+	networkClient *fakeNetwork.Clientset
+	nwInformer    cache.SharedIndexInformer
+	gnpInformer   cache.SharedIndexInformer
+	nodeInformer  coreinformers.NodeInformer
+	clusterValues gce.TestClusterValues
+	controller    *Controller
+	metrics       *controllers.ControllerManagerMetrics
+}
+
+func setupHighPerfController(ctx context.Context) *testHighPerfController {
+	fakeNetworking := fakeNetwork.NewSimpleClientset()
+
+	fakeNodeHandler := &testutil.FakeNodeHandler{
+		Existing: []*v1.Node{
+			{ObjectMeta: metav1.ObjectMeta{Name: "node0"}},
+		},
+		Clientset: fakeClient.NewSimpleClientset(),
+	}
+	fakeClient := &fakeClient.Clientset{}
+	fakeInformerFactory := informers.NewSharedInformerFactory(fakeClient, 0*time.Second)
+	nodeInformer := fakeInformerFactory.Core().V1().Nodes()
+
+	for _, node := range fakeNodeHandler.Existing {
+		nodeInformer.Informer().GetStore().Add(node)
+	}
+
+	nwInformer := networkinformer.NewNetworkInformer(fakeNetworking, 0*time.Second, cache.Indexers{})
+	gnpInformer := alphanetworkinformer.NewGKENetworkParamSetInformer(fakeNetworking, 0*time.Second, cache.Indexers{})
+	testClusterVals := gce.DefaultTestClusterValues()
+	fakeGce := gce.NewFakeGCECloud(testClusterVals)
+
+	controller := NewHighPerfController(
+		fakeGce,
+		fakeNetworking,
+		nwInformer,
+		gnpInformer,
+		nodeInformer,
+	)
+	metrics := controllers.NewControllerManagerMetrics("test")
+
+	return &testHighPerfController{
+		gceCloud:      fakeGce,
+		networkClient: fakeNetworking,
+		nwInformer:    nwInformer,
+		gnpInformer:   gnpInformer,
+		nodeInformer:  nodeInformer,
+		clusterValues: testClusterVals,
+		controller:    controller,
+		metrics:       metrics,
+	}
+}
+
+func (testVals *testHighPerfController) runHighPerfController(ctx context.Context) {
+	go testVals.nwInformer.Run(ctx.Done())
```

Comment:
@bmorettodama see here, instead of passing ctx.Done() you can pass ctx, and inside the `Run()`method, instead of passing stopCh you pass ctx.Done(), this way you don't have to create a context like the one you are creating in 

```
	ctx, cancelFn := context.WithCancel(context.Background())
	defer cancelFn()
	defer c.queue.ShutDown()
```